src/dmd/lexer.d

```diff
@@ -1824,6 +1824,13 @@ class Lexer : ErrorHandler
                 ++p;
                 base = 8;
                 break;
+            case '8':
+            case '9':
+                ++p;
+                base = 8;
+                error("radix %d digit expected, not `%c`", base, c);
+                n = n * base + (c - '0');
```

Comment:
n is zero here, so all you need is
```D
n = c - '0'.
```

Reply:
Make sure to set `err` to true also, so they don't get another error for other digits.

Essentially, I should get the same error messages for `088` as I do for `0088`

Reply:
Looking at the code here, it's very convoluted. The error messages change not based on the base, but based on the digits.

Like you get a different error message for `0b5` than you do for `0b8`.

I think there are likely better ways to structure this whole thing. But maybe too much for this PR which is focused on the `08` problem.